kernel/src/hil/sensors.rs

```diff
@@ -233,3 +234,23 @@ pub trait PressureClient {
     /// Returns the value in hPa.
     fn callback(&self, pressure: Result<u32, ErrorCode>);
 }
+
+/// Client for receiving NMEA sentances
+pub trait NmeaClient {
+    /// Called when a full NMEA sentance has been found, or if an I2C error
+    /// occurs.
+    fn callback(&self, buffer: &'static mut [u8], len: usize, status: Result<(), i2c::Error>);
+}
+
+/// A basic interface for a Nmea sentance reader
+pub trait NmeaDriver<'a> {
+    fn set_client(&self, client: &'a dyn NmeaClient);
+
+    /// Read a full NMEA sentance, including the first `$`
+    ///
+    /// When the sentance is read the `NmeaClient` callback will be called.
+    fn read_sentance(
```

Comment:
```suggestion
    fn read_sentence(
```

Should probably change this everywhere for consistency.